src/component/task/index.jsx

```diff
@@ -0,0 +1,113 @@
+import React, { useState } from "react";
+import { Link } from "react-router-dom";
+import changeIcons from "../../icons/pen.svg";
+import deleteIcons from "../../icons/trash.svg";
+import doneIcons from "../../icons/check.svg";
+import closeIcons from "../../icons/close.svg";
+import "./task.scss";
+
+const Task = ({task, onChangeTasks}) => {
+  const [isEdit, setIsEdit] = useState(false);
+  const [value, setValue] = useState(task.text);
+
+  const deleteTask = async () => {
+    try {
+      const answer = window.confirm('Are you sure?');
+      if (!answer) return;
+      await fetch(`http://localhost:8000/deleteTask?id=${task._id}`, {
+        method: "DELETE",
+      });
+      onChangeTasks();
+    } catch (e) {
+      onChangeTasks();
+    }
+  };
+
+  const updateTask = async () => {
+    try {
+      await fetch("http://localhost:8000/updateTask", {
+        method: "PATCH",
+        headers: {
+          "Content-Type": "application/json;charset=utf-8",
+          "Access-Control-Allow-Origin": "*"
+        },
+        body: JSON.stringify(task)
+      });
+      setIsEdit(false);
+      onChangeTasks();
+    } catch (e) {
+      setIsEdit(false);
+    }
+  };
+
+  const onChangeCheckbox = () => {
+    task.isCheck = !task.isCheck;
+    updateTask();
+  };
+
+  const saveUpdateText = () => {
+    task.text = value.trim();
```

Comment:
value = value.trim();
    if (!value) return alert("Enter text!");
 task.text = value;
    updateTask();

or use trim immediately when you remember the value